File: src/annotation_transfer_tool/services/transfer.py
from fast_antx.core import transfer

from annotation_transfer_tool.schemas.transfer import TransferRequest, TransferResponse


def run_transfer(request: TransferRequest) -> TransferResponse:
    patterns = [
        [label, regex]
        for label, regex in request.patterns
        if label and regex
    ]

    result = transfer(
        request.source,
        patterns,
        request.target,
        output=request.output,
    )

    return TransferResponse(result=result, output_format=request.output)

File: frontend/src/lib/api/transfer.ts
import { ApiError, parseApiError } from "@/lib/api";

export interface TransferRequest {
  source: string;
  target: string;
  patterns: [string, string][];
  output?: "txt" | "yaml" | "diff";
}

export interface TransferResponse {
  result: string | unknown[];
  output_format: string;
}

export async function transferAnnotations(
  request: TransferRequest,
): Promise<TransferResponse> {
  const response = await fetch("/api/transfer", {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({
      output: "txt",
      ...request,
    }),
  });

  if (!response.ok) {
    throw new ApiError(
      await parseApiError(response, `Transfer failed (${response.status})`),
    );
  }

  return response.json();
}
